ibis/expr/operations.py

```diff
@@ -1440,180 +1440,41 @@ class CumulativeAll(CumulativeOp):
 # ---------------------------------------------------------------------
 
 
-class TypedCaseBuilder:
-    __slots__ = ()
-
-    def type(self):
-        types = [result.type() for result in self.results]
-        return dt.highest_precedence(types)
-
-    def else_(self, result_expr):
-        """
-        Specify
-
-        Returns
-        -------
-        builder : CaseBuilder
-        """
-        kwargs = {
-            slot: getattr(self, slot)
-            for slot in self.__slots__
-            if slot != 'default'
-        }
-
-        result_expr = ir.as_value_expr(result_expr)
-        kwargs['default'] = result_expr
-        # Maintain immutability
-        return type(self)(**kwargs)
-
-    def end(self):
-        default = self.default
-        if default is None:
-            default = ir.null().cast(self.type())
-
-        args = [
-            getattr(self, slot) for slot in self.__slots__ if slot != 'default'
-        ]
-        args.append(default)
-        op = self.__class__.case_op(*args)
-        return op.to_expr()
-
-
 class SimpleCase(ValueOp):
     base = Arg(rlz.any)
-    cases = Arg(rlz.list_of(rlz.any))
-    results = Arg(rlz.list_of(rlz.any))
+    cases = Arg(rlz.value_list_of(rlz.any))
+    results = Arg(rlz.value_list_of(rlz.any))
```

Comment:
The pandas dispatcher expect a `ir.ListExpr` here produced by `ops.ValueList`.

This is the only blocker of removing `ir.ListExpr` and `ops.ValueList` in favor of a plain python list of ibis expressions.

Reply:
`list` [is not a computable](https://github.com/ibis-project/ibis/blob/master/ibis/backends/pandas/core.py#L285) arg [neither it is hashable](https://github.com/ibis-project/ibis/blob/master/ibis/backends/pandas/core.py#L337)

Reply:
One option might be to covert everything into `tuple`s at the API boundary and use `tuple` as the internal representation.

Reply:
I tried that in a different attempt, though hit a couple of roadblocks. Going to try it again in a separate PR.